src/core/mesh/qgsmeshtracerenderer.h

```diff
@@ -0,0 +1,461 @@
+/***************************************************************************
+                         qgsmeshtracerenderer.h
+                         -------------------------
+    begin                : November 2019
+    copyright            : (C) 2019 by Vincent Cloarec
+    email                : vcloarec at gmail dot com
+ ***************************************************************************/
+
+/***************************************************************************
+ *                                                                         *
+ *   This program is free software; you can redistribute it and/or modify  *
+ *   it under the terms of the GNU General Public License as published by  *
+ *   the Free Software Foundation; either version 2 of the License, or     *
+ *   (at your option) any later version.                                   *
+ *                                                                         *
+ ***************************************************************************/
+
+#ifndef QGSMESHTRACERENDERER_H
+#define QGSMESHTRACERENDERER_H
+
+#define SIP_NO_FILE
+
+#include <QVector>
+#include <QSize>
+
+#include "qgis_core.h"
+#include "qgis.h"
+#include "qgstriangularmesh.h"
+#include "qgsmeshlayer.h"
+#include "qgsmeshlayerutils.h"
+
+
+///@cond PRIVATE
+
+/**
+ * \ingroup core
+ *
+ * Abstract class used to interpolate the value of the vector for a pixel
+ *
+ * \note not available in Python bindings
+ * \since QGIS 3.12
+ */
+class CORE_EXPORT QgsMeshVectorValueInterpolator
+{
+  public:
+    //! Constructor
+    QgsMeshVectorValueInterpolator( QgsTriangularMesh triangularMesh,
+                                    QgsMeshDataBlock datasetVectorValues,
+                                    QgsMeshDataBlock scalarActiveFaceFlagValues );
+    //! Destructor
+    virtual ~QgsMeshVectorValueInterpolator();
+
+    //! Returns the interpolated vector
+    virtual QgsVector value( const QgsPointXY &point ) = 0;
+
+  protected:
+    void updateFace( const QgsPointXY &point );
+
+    QgsTriangularMesh mTriangularMesh;
+    QgsMeshDataBlock mDatasetValues;
+    QgsMeshDataBlock mActiveFaceFlagValue;
+    QgsMeshFace face;
+    int mCacheFaceIndex;
+};
+
+
+/**
+ * \ingroup core
+ *
+ * Class used to retrieve the value of the vector for a pixel from vertex
+ *
+ * \note not available in Python bindings
+ * \since QGIS 3.12
+ */
+class  CORE_EXPORT QgsMeshVectorValueInterpolatorFromVertex: public QgsMeshVectorValueInterpolator //CORE_EXPORT needed to no have v-table error ????
```

Comment:
probably most of the classes in this header does not need to be exported from qgis_core library (remove CORE_EXPORT for such classes)